include/ocpp/v201/smart_charging.hpp

```diff
@@ -45,6 +47,42 @@ enum class ProfileValidationResultEnum {
     DuplicateProfileValidityPeriod
 };
 
+/// \brief This enhances the ChargingProfile type by additional paramaters that are required in the
+/// ReportChargingProfilesRequest (EvseId, ChargingLimitSourceEnum)
+struct ReportedChargingProfile : public ChargingProfile {
```

Comment:
Why derive from ChargingProfile? Why not use composition, since this struct seems to do nothing more than provide two other data members, and you provide the ability to get back the original profile anyway. Instead, you can make the ChargingProfile a member.